pysnooper/tracer.py

```diff
@@ -223,9 +223,23 @@ def __init__(
         self.custom_repr = custom_repr
         self.last_source_path = None
 
-    def __call__(self, function):
+    def __call__(self, function_or_class):
         if DISABLED:
             return function
+
+        if inspect.isclass(function_or_class):
+            return self._wrap_class(function_or_class)
+        else:
+            return self._wrap_function(function_or_class)
+
+    def _wrap_class(self, cls):
+        for attr_name in dir(cls):
```

Comment:
This will snoop methods from base classes, which again can lead to logs from snoop even when the decorated class isn't being used. I suggest using `__dict__` to only touch methods defined directly in the current class.

Reply:
Agreed.